app/views/collections/show.html.erb

```diff
@@ -0,0 +1,186 @@
+    data-uid="<%= user.uid %>"
+    data-collection-id="<%= @collection.id %>"
+    data-can-delete="<%= @can_edit && user.uid != current_user.uid %>"
+    data-you="<%= user.uid == current_user.uid %>"></span>
+<% end %>
+</ul>
+
+<% if @can_edit == true %>
+  <div>
+      <input id="submitter-uid-to-add" placeholder="netid"></input>
+      <%= link_to "Add Submitter", "#", id: "btn-add-submitter", class: "btn btn-secondary" %>
+      <span id="add-submitter-message"></span>
+  </div>
+<% end %>
+
+
+<h2>Administrators</h2>
+<p>Users that are allowed to admininister this collection</p>
+<ul id="admin-list">
+<% @collection.administrators.each do |user| %>
+  <span class="hidden admin-data"
+    data-uid="<%= user.uid %>"
+    data-collection-id="<%= @collection.id %>"
+    data-can-delete="<%= @can_edit && user.uid != current_user.uid %>"
+    data-you="<%= user.uid == current_user.uid %>"></span>
+<% end %>
+</ul>
+
+<% if @can_edit == true %>
+  <div>
+      <input id="admin-uid-to-add" placeholder="netid"></input>
+      <%= link_to "Add Administrator", "#", id: "btn-add-admin", class: "btn btn-secondary" %>
+      <span id="add-admin-message"></span>
+  </div>
+<% end %>
+
+<% if @can_edit == true %>
+  <div>
+    <%= link_to("Edit", edit_collection_path(@collection), class: "btn btn-primary") %>
+  </div>
+<% end %>
+
+<%= link_to("View all collections", collections_path) %>
+
+<script>
+$(function() {
+
+  // Issues the HTTP DELETE to remove a user's access from a collection
+  var deleteUserFromCollection = function(el, url, uid) {
+    $.ajax({
+      type: "DELETE",
+      url: url.replace("uid-placeholder", uid),
+      data: { authenticity_token: "<%= form_authenticity_token %>" },
+      success: function() {
+        // Remove the <li> for the user
+        $(".li-user-" + uid).remove();
+      },
+      error: function(x) {
+        alert(x.responseJSON.message);
+      }
+    });
+  }
+
+  // Adds the information for a given user to the lists of administrators/submitters
+  // for the collection.
+  // `elList` is the reference to the <ul> HTML element that hosts the list.
+  var addUserHtml = function(elList, uid, collectionId, role, canDelete, isYou) {
+    var base_user_url = "<%= user_path('user-placeholder') %>";
+    var show_user_url = base_user_url.replace("user-placeholder", uid);
+    var html = `<li class="li-user-${uid}"> <a href="${show_user_url}">${uid}</a>`;
+    if (isYou == true) {
+      html += ` (you)`;
+    }
+    if (canDelete == true) {
+      html += `
+        <span>
+        <a class="delete-user-from-collection" data-collection-id="${collectionId}" data-uid="${uid}"
+          href="#" title="Revoke user's ${role} access this collection">
+          <i class="bi bi-trash delete_icon" data-collection-id="${collectionId}" data-uid="${uid}"></i>
+        </a>
+      </span>`;
+    }
+    $(elList).append(html);
+  }
+
+  // Issues the HTTP POST to add a user to a collection and updates the UI
```

Comment:
Adding and removing submitters and administrators to the collection is done via AJAX calls, so the screen does not jump/refresh every time a new submitter or administrator is added. These options are only available when the user has the proper admin rights to the collection.